src/DynamicProbes.csproj

```diff
@@ -7,6 +7,18 @@
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+
+    <!-- Persist the source generator (and other) files to disk -->
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+    <!-- The base path for the generated source files -->
+    <GeneratedFolder>Generated</GeneratedFolder>
+    <!-- Write the output for each target framework to a different subfolder -->
+    <CompilerGeneratedFilesOutputPath>$(GeneratedFolder)/$(TargetFramework)</CompilerGeneratedFilesOutputPath>
   </PropertyGroup>
 
+<ItemGroup>
+    <!-- Exclude the output of source generators from the compilation to avoid the ambiguous definition errors (CS0121) -->
+    <Compile Remove="$(GeneratedFolder)/**/*.cs" />
+</ItemGroup>
```

Comment:
nit: indentation:

```suggestion
  <ItemGroup>
    <!-- Exclude the output of source generators from the compilation to avoid the ambiguous definition errors (CS0121) -->
    <Compile Remove="$(GeneratedFolder)/**/*.cs" />
  </ItemGroup>
```